src/parser/style-context.ts

```diff
@@ -0,0 +1,94 @@
+import type { Node, Parser, Root } from "postcss";
+import postcss from "postcss";
+import { parse as SCSSparse } from "postcss-scss";
+
+import type { Context } from "../context";
+import type { SourceLocation, SvelteStyleElement } from "../ast";
+
+export interface StyleContext {
+  sourceLang: string | null;
+  sourceAst: Root | null;
+}
+
+/**
+ * Extracts style source from a SvelteStyleElement and parses it into a PostCSS AST.
+ */
+export function parseStyleContext(
+  styleElement: SvelteStyleElement | undefined,
+  ctx: Context
+): StyleContext {
+  const styleContext: StyleContext = {
+    sourceLang: null,
+    sourceAst: null,
+  };
+  if (!styleElement || !styleElement.endTag) {
+    return styleContext;
+  }
+  styleContext.sourceLang = "css";
+  for (const attribute of styleElement.startTag.attributes) {
+    if (
+      attribute.type === "SvelteAttribute" &&
+      attribute.key.name === "lang" &&
+      attribute.value.length > 0 &&
+      attribute.value[0].type === "SvelteLiteral"
+    ) {
+      styleContext.sourceLang = attribute.value[0].value;
+    }
+  }
+  let parseFn: Parser<Root>;
+  switch (styleContext.sourceLang) {
+    case "css":
+      parseFn = postcss.parse;
+      break;
+    case "scss":
+      parseFn = SCSSparse;
+      break;
+    default:
+      return styleContext;
+  }
+  const styleCode = ctx.code.slice(
+    styleElement.startTag.range[1],
+    styleElement.endTag.range[0]
+  );
+  try {
+    styleContext.sourceAst = parseFn(styleCode, {
+      from: ctx.parserOptions.filePath,
+    });
+    // eslint-disable-next-line no-empty -- Catching errors is not a good way to go around this, a safe parser should be used instead.
+  } catch {}
+  fixPostCSSNodeLocation(
+    styleContext.sourceAst,
+    styleElement.loc,
+    styleElement.startTag.range[1]
+  );
+  styleContext.sourceAst.walk((node) =>
+    fixPostCSSNodeLocation(
+      node,
+      styleElement.loc,
+      styleElement.startTag.range[1]
+    )
+  );
+  return styleContext;
+}
+
+/**
+ * Fixes PostCSS AST locations to be relative to the whole file instead of relative to the <style> element.
+ */
+function fixPostCSSNodeLocation(
+  node: Node,
+  styleElementLoc: SourceLocation,
+  styleElementOffset: number
+) {
+  if (node.source?.start?.offset !== undefined) {
+    node.source.start.offset += styleElementOffset;
+  }
+  if (node.source?.start?.line !== undefined) {
```

Comment:
For `<style>` tags like below, I think we also need to change the `column` position.

```svelte
<style> a { color: red; } </style>
```

Reply:
Oh no...

Reply:
However I did fix the issue you reported. I just don't know if it even makes sense to try to offset things... :/

Reply:
Thanks for your opinion.
As you said, I think the difference in range specifications can be confusing.
However, I don't want to write calculation logic each time a new rule in the future requests to create a report from a PostCSS node.
What do you think about providing a helper function to get the report range from PostCSS nodes via the parser service instead of reassigning all node positions?

Reply:
That's a good idea, I have implemented a pair of functions that does that and tested them.

However, I don't know how to export them - should they be a normal export from the package? So you would (as a user) do just 
```ts
import { styleNodeLoc, styleNodeRange } from "svelte-eslint-parser";

styleNodeLoc(services.getStyleContext().sourceAst!);
```
Or should they be properties on `StyleContext`? So you would do
```ts
const styleContext = services.getStyleContext();
styleContext.styleNodeLoc(styleContext.sourceAst!)
```

Which do you think is better? I noticed that you don't export many things directly...

Also, should the `StyleContext` type (or the subtypes) be exported?

Reply:
Sorry for the late reply. I've been busy with office work lately 😓 
I was thinking of providing it from a parser service. So the second example is the same.

Reply:
> Also, should the `StyleContext` type (or the subtypes) be exported?

That's a good point. I think it makes sense to expose `StyleContext` type so that we can write rules in TypeScript.

Reply:
No worries about the time :)

I have added `styleNodeLoc` and `styleNodeRange` to the parser services and exported the types around `StyleContext`.